_layouts/ballot.html

```diff
@@ -4,13 +4,12 @@
 {% assign ballot = page %}
 {% assign locality = site.localities | where: 'slug', ballot.locality | first %}
 
-{% assign ballots = site.elections | where: "locality", locality.locality_id | reverse %}
-{% assign current_ballot = ballots | where: "election", locality.current_election | first %}
 {% assign today = site.time | date: "%Y-%m-%d" %}
 {% assign current_election = locality.current_election | date: "%Y-%m-%d" %}
-{% assign page.ballot = ballot %}
+{% assign ballot_year = page.election | slice: 0, 4 %}
+{% assign ballots = site.elections | where_exp: 'ballot', 'ballot.election contains ballot_year' %}
```

Comment:
Cool, I see what you're doing here - it makes sense per your comment on how you're hacking around the current data structure. It just pains me to see the same logic duplicated in each `layout`... Is there a way we can make these two values global? I have no idea how to do that though 🙂 Maybe this is the best we've got for now, since it also looks like there's a similar pattern in the `layouts` where there is the same data being `assign`ed in multiple files.

Reply:
I simplified it: moved into ballot-nav coz only ballot-nav that needs all ballots for the year. 🙂

Reply:
Sweet! 🍬

Reply:
Although I still see this in a bunch of other places - is there something I'm missing?

Reply:
Ah, I see I missed one in ballot.html. The others, in locality.html and all-elections.html, are for something else. I didn't look into those files.